### PR TITLE
Item grid component to display items in a grid.

### LINKED ISSUE #174: Item Grid Component
The ItemGrid core component should be moved into Lakefront and then replace usage in Lakehouse for easier maintenance and better documentation.

### PR DESCRIPTION
**Pull request by NehalBhandariSlalom**
This PR is for Item Grid Component that displays items in a grid. If approve, this will close #174.

Screen Shot 2021-10-05 at 10 14 17 AM

### CODE REVIEW

File: src/ItemGrid/ItemGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import React, { useCallback, useState } from 'react';
+import { ItemGridContainer } from './itemGridStyles';
+
+export interface ItemGridProps {
+    /**
+     * This is the max number of columns that populates in a row.
+     */
+    maxColumns: number;
+    /**
```
Comment (amirkibbar) [inline src/ItemGrid/ItemGrid.tsx:9]: nit: can you add a blank line above all appearances of `/**`?

File: src/ItemGrid/index.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import ItemGrid, { ItemGridProps } from './ItemGrid';
+
+export { ItemGridProps };
+export default ItemGrid;;
```
Comment (amirkibbar) [inline src/ItemGrid/index.ts:4]: typo: extra `;`

**Review by amirkibbar — APPROVED** [review]
but please fix the two small things

File: src/ItemGrid/index.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import ItemGrid, { ItemGridProps } from './ItemGrid';
+
+export { ItemGridProps };
+export default ItemGrid;
```
Comment (jason-ransom-slalom) [inline src/ItemGrid/index.ts:4]: Add a new line at the end of the file.

File: src/ItemGrid/index.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import ItemGrid, { ItemGridProps } from './ItemGrid';
+
+export { ItemGridProps };
```
Comment (jason-ransom-slalom) [inline src/ItemGrid/index.ts:3]: We also need to export in the src/index (for consumer use) and add a link to it in the README.md

**Review by jason-ransom-slalom — CHANGES_REQUESTED** [review]

File: src/ItemGrid/index.ts (hunk 4 above)
Comment (NehalBhandariSlalom) [reply src/ItemGrid/index.ts:3]: Completely forgot about that part. Thanks for reminding.

**Review by jason-ransom-slalom — APPROVED** [review]

**Pull request merged**